benchmarks/graphics_pipeline/main.cpp

```diff
@@ -701,49 +454,296 @@ void ProjApp::Setup()
+        RepeatGeometryNonPositionVertexData(highPrecisionInterleavedSingleSphere, kMaxSphereInstanceCount, highPrecisionInterleaved);
+        RepeatGeometryNonPositionVertexData(highPrecisionPositionPlanarSingleSphere, kMaxSphereInstanceCount, highPrecisionPositionPlanar);
+
+        // Get position buffers and prepare to edit in place
+        Geometry::Buffer* lowInterleavedPositionBufferPtr  = lowPrecisionInterleaved.GetVertexBuffer(0);
+        Geometry::Buffer* lowPlanarPositionBufferPtr       = lowPrecisionPositionPlanar.GetVertexBuffer(0);
+        Geometry::Buffer* highInterleavedPositionBufferPtr = highPrecisionInterleaved.GetVertexBuffer(0);
+        Geometry::Buffer* highPlanarPositionBufferPtr      = highPrecisionPositionPlanar.GetVertexBuffer(0);
+
+        // Resize empty Position Planar vertex buffers
+        lowPlanarPositionBufferPtr->SetSize(sphereVertexCount * kMaxSphereInstanceCount * lowPlanarPositionBufferPtr->GetElementSize());
+        highPlanarPositionBufferPtr->SetSize(sphereVertexCount * kMaxSphereInstanceCount * highPlanarPositionBufferPtr->GetElementSize());
+
+        // Iterate through the full vertex buffers, changing position data and appending indices.
+        //
+        // i : sphere index
+        // j : vertex index within one sphere
+        // k : triangle index within one sphere
+        // v0, v1, v2: the three elements of triangle k
+        //
+        // Full vertex buffers contain a total of (kMaxSphereInstanceCount * sphereVertexCount) vertices arranged like so:
+        //
+        // | j(0) | j(1) | ... | j(sphereVertexCount-1) | j(0) | j(1) | ... | j(sphereVertexCount-1) | ... | j(0) | j(1) | ...  | j(sphereVertexCount-1) |
+        // |--------------------i(0)--------------------|--------------------i(1)--------------------| ... |----------i(kMaxSphereInstanceCount)---------|
+        //
+        // Full index buffers contain a total of (kMaxSphereInstanceCount * sphereTriCount * 3) indices arranged like so:
+        //
+        // | v0 | v1 | v2 | v0 | v1 | v2 |    ...     | v0 | v1 | v2 | ... | v0 | v1 | v2 | v0 | v1 | v2 |    ...     | v0 | v1 | v2 |
+        // |     k(0)     |     k(1)     | ... | k(sphereTriCount-1) | ... |     k(0)     |     k(1)     | ... | j(sphereTriCount-1) |
+        // |---------------------------i(0)--------------------------| ... |--------------i(kMaxSphereInstanceCount-1)---------------|
+        //
+        for (uint32_t i = 0; i < kMaxSphereInstanceCount; i++) {
+            uint32_t index = sphereIndices[i];
+            uint32_t x     = (index % (grid.xSize * grid.ySize)) / grid.ySize;
+            uint32_t y     = index % grid.ySize;
+            uint32_t z     = index / (grid.xSize * grid.ySize);
+
+            // Model matrix to be applied to the sphere mesh
+            float4x4 modelMatrix = glm::translate(float3(x * grid.step, y * grid.step, z * grid.step));
```

Comment:
I think even this can be a small helper function: GetModelMatrix()

Reply:
Changed, refactored a grid class